DESIGN.md

```diff
@@ -0,0 +1,205 @@
+
+A `Chain` is a specific blockchain, potentially with overrides for slight differences in the platform implementation. 
+
+A `Protocol` (fka `Module`) is a specific application on a `Chain`, it provides a set of methods that can be called to accomplish some action (e.g. `TokenBridge` allows send/receive/lookup token, etc...)
+
+A `Signer` is an interface that provides a callback to sign one or more transaction objects. These signed transactions are sent to the blockchain to invoke some action.
+
+An `Attestation` is a signed guarantee that some _thing_ happened on a remote chain, sent to the target chain to complete a transfer.
+
+
+# Details 
+
+## Wormhole 
+
+Registers Platforms
+
+Allows overriding chain specific configs (rpc, contract addresses, ...)
+
+Provides methods to get PlatformContext or ChainContext objects
+```ts
+wh.getPlatform("Evm")
+wh.getChain("Ethereum")
+```
+
+Provides methods to create a `WormholeTransfer` for any `Protocol`
+```ts
+wh.tokenTransfer(...)
+wh.nftTransfer(...)
+wh.cctp(...)
+//...
+```
+
+Provides methods to query an API for VAAs and token details
+```ts
+// grab a vaa with identifier
+wh.getVaa(...)
+// get the token details 
+wh.getOriginalToken(...)
+wh.getWrappedToken(orig, chain)
+```
+
+## Platform
+
+Base class, implements Platform specific logic
+
+Parse Addresses
+
+Parse Message out of Transaction
+
+Sign/Send/Wait
+
+## ChainContext
+
+Defined in abstract ChainContext class
+
+Note: Dispatches many calls to the Platform, filling in details like ChainName and RPC
+
+The `getRpc` method is the only platform specific thing _required_ to implement.
+
+Responsible for holding RPC connection, initialized from default or overrides
+```ts
+cc.getRPC() // for evm -> ethers.Provider, for sol -> web3.Connection
+```
+
+Holds references to Contract client 
+
+
+<!-- 
+Not Implemented
+Provides methods to lookup details for contract addresses, finality, address parsers/formatters
+
+```ts
+cc.getTokenBridgeAddress()
+cc.estimateFinality(txid)
+```
+-->
+
+
+## WormholeTransfer
+
+Holds a reference to ChainContexts
+
+Holds details about the transfer
+
+Provides methods to step through the transfer process
+
+## Glossary
+
+- Network
+    Mainnet, Testnet, Devnet
+- Platform
+    A chain or group of chains within the same ecosystem that share common logic (e.g. EVM, Cosmwasm, etc)
+- Platform Context
+    A class which implements a standardized format and set of methods. It may include additional chain-specific methods and logic.
+- Protocol 
+    A cross-chain application built on top of Wormhole (the core contracts are also considered a module)
+- Universal Address
+    A 32-byte address, used by the wormhole contracts
+- Native Address (I think this should be called "Platform Address")
+    An address in the standard chain-specific format
+- Native
+    The "home" chain (e.g. ETH is native to Ethereum)
```

Comment:
good point/agree re Platform Address

I guess if we want to use the word "native", we should have some coherent idea what it should represent, because right now it's quite the confusing grab-bag of different concepts (e.g. calling the gas token of a chain the native token, calling the origin token of the token bridge the native / home chain etc.)